### PR TITLE
Stops ice demons from SHATTERING the game's balance by dying. (Fixes infinite activated bluespace anomaly core farms)

### LINKED ISSUE #55229: Ice demon drops ignore anomaly core type limits
## Round ID:
Tested locally

## Testmerges:
N/A

## Reproduction:
Kill an ice demon, and get a 5% to drop a new anomaly core that doesn't check or update created_anomaly_types in the research ss. 
https://github.com/tgstation/tgstation/blob/6b6275738f4c3db590e74fc5da8590d309e2b72e/code/modules/mob/living/simple_animal/hostile/mining_mobs/ice%20demon.dm#L82

This means the "hard limit" of 8 in the round can be circumvented, and the anomaly core system isn't being used to control the maximum number of end game items available.

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
This replaces the activated core dropped by the ice demon with a raw core, and replaces the activated core sometimes dropped by their portal with 23 bags of cheese honkers, three empty.
Closes #55229

## Why It's Good For The Game
Anomaly core limit bypasses are bad.

## Changelog
:cl:
fix: Ice demons and their portals don't drop activated anomaly cores anymore
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
cheese honker moment

**Comment by optimumtact** [comment]
this is two strikes for meme pr's, another and you're done.

why would it spawn a bunch of cheesey honkers

File: code/game/objects/structures/icemoon/cave_entrance.dm [hunk 1]
```diff
@@ -169,7 +169,30 @@ GLOBAL_LIST_INIT(ore_probability, list(
...
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
+			new /obj/item/trash/cheesie(loc)
```
Comment (ATH1909) [inline code/game/objects/structures/icemoon/cave_entrance.dm:195]:
```suggestion
			new /obj/item/raw_anomaly_core/bluespace(loc)
```

Comment (ATH1909) [reply code/game/objects/structures/icemoon/cave_entrance.dm:195]: I'd suggest adding a check to see if the anomaly core cap has been reached and dropping an activated core if it hasn't, but I'm a bit worried that a miner (or even just a portal or demon somehow dying on their own) could "soak up" parts of the bluespace anomaly core limit accidentally.

**Comment by ATH1909** [comment]
tbh, is it really _that_ bad if one or two extra bluespace anomaly cores get spawned?

**Comment by optimumtact** [comment]
yes it is bad

**Comment by interestingusernam3** [comment]
>this is two strikes for meme pr's, another and you're done.

>why would it spawn a bunch of cheesey honkers

An inactive anomaly core is trash for a miner so instead of downgrading the active core I decided to replace that option with something that's a bit funny instead of just being a dissapointment.

As for the lore... the demons got to eat something and cheese honkers are too good to be produced with mundane machinery, so this closes two plot holes.

Also this isn't a meme PR, I unironically want this merged.

**Comment by interestingusernam3** [comment]
I buffed this a little, now it makes 20 full bags and 3 empty ones so the drop is actually usable as a food source.

**Comment by Timberpoes** [comment]
If you want to re-submit this without the cheese honkers change, feel free.

image

**Comment by interestingusernam3** [comment]
Fine.

**Pull request closed without merging**